### DISCUSSION TITLE
Can authenticated users see function definitions?

### DISCUSSION THREAD

**Discussion by Oreoezi** [supabase/supabase#21513] [Questions]
Doing 

```sql
SELECT proname, prosrc
FROM pg_proc
WHERE proname NOT LIKE 'pg_%';

SELECT proname, pg_get_functiondef(oid) AS definition
FROM pg_proc
WHERE proname = 'my_function_here';
```

as a mock aunthenticated user from the SQL Editor in the Supabase dashboard, I can see the code of a function. That can be problematic if the function calls an API and therefore leaks credentials. I tried defining the function in another schema and that didn't seem to help. Is this only possible due to limitations with the SQL editor or is it a genuine matter of concern?

**Comment by GaryAustin1** [comment]
If you mean access from the API, you can't get to that table.
